### PR TITLE
[schemaDB] add checkpoint support

### PR DESCRIPTION
**Pull request by lightmark**
## Motivation

add checkpoint support for better benchmarking

### Have you read the [Contributing Guidelines on pull requests](https://github.com/diem/diem/blob/main/CONTRIBUTING.md#pull-requests)?

Y

## Test Plan

add ut.

### CODE REVIEW

File: storage/schemadb/tests/db.rs [hunk 1]
```diff
@@ -384,3 +384,22 @@ fn test_report_size() {
...
+    }
+    {
+        let db = open_db(&checkpoint);
+        assert_eq!(
+            db.get::<TestSchema1>(&TestField(0)).unwrap(),
+            Some(TestField(0)),
+        );
+        db.put::<TestSchema1>(&TestField(1), &TestField(1)).unwrap();
+    }
+}
```
Comment (msmouse) [inline storage/schemadb/tests/db.rs:405]: Test that writing to the checkpoint didn't affect the original DB (by reading 0 and 1)?

File: storage/schemadb/tests/db.rs [hunk 2]
```diff
@@ -384,3 +384,29 @@ fn test_report_size() {
...
+            Some(TestField(0)),
+        );
+
+        let cp = open_db(&checkpoint);
+        assert_eq!(
+            cp.get::<TestSchema1>(&TestField(0)).unwrap(),
+            Some(TestField(0)),
+        );
+        cp.put::<TestSchema1>(&TestField(1), &TestField(1)).unwrap();
+        assert_eq!(db.get::<TestSchema1>(&TestField(1)).unwrap(), None);
```
Comment (wqfish) [inline storage/schemadb/tests/db.rs:410]: I'd also read 1 from the checkpoint.

**Review by wqfish — APPROVED** [review]

**Review by wqfish — DISMISSED** [review]

**Pull request closed without merging**